lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -294,6 +300,23 @@ def check_if_recordname_exists(self, obj_filter, ib_obj_ref, ib_obj_type, curren
             if obj_host_name == ref_host_name and current_ip_addr != proposed_ip_addr:
                 self.create_object(ib_obj_type, proposed_object)
 
+    def check_if_nios_next_ip_exists(self, proposed_object):
+        ''' Check if nios_next_ip argument is passed in ipaddr while creating
+            host record, if yes then format proposed object ipv4addrs and pass
+            func:nextavailableip and ipaddr range to create hostrecord with next
+             available ip in one call to avoid any race condition '''
+
+        if 'ipv4addrs' in proposed_object:
+            if 'nios_next_ip' in proposed_object['ipv4addrs'][0]['ipv4addr']:
+                ip_range = ast.literal_eval(proposed_object['ipv4addrs'][0]['ipv4addr']).keys()[1]
```

Comment:
Can you explain what's going on here?

Reply:
So, here `proposed_object` will have dict as: `'ipv4addrs': [{'ipv4addr': "{'nios_next_ip': None, '192.168.11.0/24': None}"}]` and above logic is to get to `192.168.11.0/24` i.e. ip_range in which next_available_ip will be searched.
Brief context:
previous to this logic, dynamic ip allocated to host record use to done like:
```
ipv4:
   - address: "{{ lookup('nios_next_ip', '192.168.11.0/24', provider={nios_provider}"
```
but the issue with the logic was a race condition as two seperate calls were being made one to lookuo and another to set address to host_record as mentioned in the bug. And now, it can be implemented as:
```
ipv4:
   - address: {nios_next_ip, 192.168.11.0/24}
```
which ultimately resolves the race condition (ref: https://community.infoblox.com/t5/DNS-DHCP-IPAM/Simultaneous-requests-for-next-available-IP/td-p/6617)

Reply:
I guess my question is why does it come back like that? Importing `ast` is generally a huge red flag for me (even for just `literal_eval`), and I see why you've done it, but is there a better way of generating this value that doesn't involve, well, this?